File: sdks/java/src/main/java/dev/atap/sdk/exception/ATAPConflictException.java
package dev.atap.sdk.exception;

import dev.atap.sdk.model.ProblemDetail;

/**
 * Conflict error (409).
 */
public class ATAPConflictException extends ATAPException {

    private final ProblemDetail problem;

    public ATAPConflictException(String message, ProblemDetail problem) {
        super(message, 409);
        this.problem = problem;
    }

    public ATAPConflictException(String message) {
        this(message, null);
    }

    public ProblemDetail getProblem() {
        return problem;
    }
}

File: sdks/php/src/Exception/ATAPConflictException.php
<?php

declare(strict_types=1);

namespace Atap\Sdk\Exception;

use Atap\Sdk\Model\ProblemDetail;

/**
 * Conflict error (409).
 */
class ATAPConflictException extends ATAPException
{
    public function __construct(
        string $message = 'Conflict',
        public readonly ?ProblemDetail $problem = null,
        ?\Throwable $previous = null,
    ) {
        parent::__construct($message, 409, $previous);
    }
}
